### PR TITLE
Replace ParcelableCrdtData and ParcelableCrdtOperation with protos

### PR DESCRIPTION
**Pull request by csilvestrini**
This means I can finally delete all the Parcelable classes for all the crdt subtypes

### CODE REVIEW

File: java/arcs/android/crdt/CrdtParcelables.kt [hunk 1]
```diff
@@ -12,95 +12,42 @@
...
-        requireNotNull(expectedType.crdtOperationCreator) {
-            "No crdtOperationCreator for type $expectedType"
-        }
-    )?.actual
+fun Parcel.readOperation(): CrdtOperation? =
+    readProto(CrdtOperationProto.getDefaultInstance())?.toOperation()
 
 /** Writes a [List] of [CrdtOperation]s to the [Parcel]. */
-fun Parcel.writeOperations(operations: List<CrdtOperation>, flags: Int) {
+fun Parcel.writeOperations(operations: List<CrdtOperation>) {
```
Comment (galganif) [inline java/arcs/android/crdt/CrdtParcelables.kt:39]: have you considered having multiple operations inside a single CrdtOperationsProto (instead then CrdtOperationProto containing only one op)?

File: java/arcs/android/crdt/CrdtParcelables.kt [hunk 2]
```diff
@@ -12,95 +12,42 @@
...
-            writeTypedObject((operation as? CrdtSet.Operation<Referencable>)?.toParcelable(), flags)
-        is CrdtSingleton.Operation<*> ->
-            writeTypedObject(
-                (operation as? CrdtSingleton.Operation<Referencable>)?.toParcelable(), flags)
-        is CrdtEntity.Operation ->
-            TODO("Implement me when ParcelableEntity is ready")
-        else ->
-            throw IllegalArgumentException("Unsupported CrdtOperation type: ${operation.javaClass}")
-    }
+fun Parcel.writeOperation(operation: CrdtOperation?) {
```
Comment (galganif) [inline java/arcs/android/crdt/CrdtParcelables.kt:30]: can this be non-nullable?

File: java/arcs/android/crdt/CrdtParcelables.kt (hunk 1 above)
Comment (csilvestrini) [reply java/arcs/android/crdt/CrdtParcelables.kt:39]: It's only used in once place, and I'm hoping to get rid of that usage anyway

File: java/arcs/android/crdt/CrdtParcelables.kt (hunk 2 above)
Comment (csilvestrini) [reply java/arcs/android/crdt/CrdtParcelables.kt:30]: Yep, good call!

**Pull request merged**